tests/basic_tests.cpp

```diff
@@ -17,6 +17,7 @@ TYPED_TEST(basic_tests, insane_url) {
   ASSERT_EQ(r->get_hostname(), "");
   ASSERT_EQ(r->get_port(), "");
   ASSERT_EQ(r->get_pathname(), "@EEEEEEEEEE");
```

Comment:
This will help you catch more bugs:

```suggestion
  ASSERT_EQ(r->get_pathname(), "@EEEEEEEEEE");
  ASSERT_TRUE(r->is_valid);
```